cmd/devicesimulator/main.go

```diff
@@ -233,101 +288,167 @@ func createAgentConfigTemplate(dataDir string, configFile string) *agent_config.
 		log.Fatalf("Error validating config: %v", err)
 	}
 
-	return agentConfigTemplate
+	// if the config does not have an explicit log level we can override it
+	return agentConfigTemplate, tmpConfig.LogLevel == ""
```

Comment:
There is no good reason to read the configuration value here and to return a boolean flag which is used to override a structure (agentConfigTemplate) that is created here.  If you decided that you need to override it - then override it here.
BTW, aren't there other values that might be overridden by the configuration file ?